### PR TITLE
Detach the player from entities on death.

### LINKED ISSUE #5075: Players killed while attached to entity stay attached and don't respawn normaly.
E.g if a player is in a boat and gets killed by TNT they will respawn still in the boat in the same place.

### PR DESCRIPTION
**Pull request by red-001**
fixes #5075.

### CODE REVIEW

**Comment by pilino1234** [comment]
If you die while in a boat, you'll respawn sitting down until you move. Also, if you died on a boat in water by lava running over you, and lavacooling creates stone over the boat where you died, you will spawn with the lighting of the boat under the stone. Once you dig the stone, your player will become normally coloured again:
screenshot_20170121_105609
screenshot_20170121_105740


Also, once you respawn you still control the boat, which is quite funny. So, if you're walking around, the boat will drive too.

File: src/content_sao.cpp [hunk 1]
```diff
@@ -204,6 +204,16 @@ void UnitSAO::getAttachment(int *parent_id, std::string *bone, v3f *position,
 	*rotation = m_attachment_rotation;
 }
 
+void UnitSAO::detach()
```
Comment (nerzhul) [inline src/content_sao.cpp]: That make sense, but can you rename it to detachFromParent instead of too generic name?

Comment (red-001) [inline src/content_sao.cpp]: done.

**Comment by nerzhul** [comment]
Approved by @sfan5  on IRC

**Comment by pilino1234** [comment]
Did nobody read my comment? This is completely broken @nerzhul @sfan5 @red-001 Why was this merged as-is?
* The player respawns in previous position (sitting in case of boat)
* The player model is still subject to the lighting of the entity it was attached to (screenshots above)

**Pull request merged**